ql/src/java/org/apache/hadoop/hive/ql/Context.java

```diff
@@ -1225,8 +1228,14 @@ public Table getMaterializedTable(String cteName) {
     return cteTables.get(cteName);
   }
 
-  public void addMaterializedTable(String cteName, Table table) {
+  public void addMaterializedTable(String cteName, Table table, Statistics statistics) {
     cteTables.put(cteName, table);
+    cteTables.put(table.getFullyQualifiedName(), table);
```

Comment:
I moved this logic here for easiness, which means this pair is stored for SemanticAnalyzer. I would say the overhead is trivial and acceptable. Anyway, I hope most people use CalcitePlanner.